lang/java/avro/src/test/java/org/apache/avro/TestSchemaCompatibility.java

```diff
@@ -260,6 +261,10 @@ public void testUnionReaderWriterSubsetIncompatibility() {
 
       new ReaderWriter(ENUM1_AB_SCHEMA, ENUM1_AB_SCHEMA), new ReaderWriter(ENUM1_ABC_SCHEMA, ENUM1_AB_SCHEMA),
```

Comment:
We should probably add an enum test here that has symbol in the write schema that are not in the read schema, but where the read schema has a default symbol.

Reply:
For this PR do you mean? It seems a a bit separate from the issue I'm working on, though it'd be easy enough to add such a test while I'm working in this area.

Reply:
True, it has no relation to the current PR. But as we're touching the code, we may just as well apply the boyscout rule.

Reply:
Makes sense - included a fix for this in my latest changes